src/ecore.js

```diff
@@ -861,6 +868,9 @@ EClass.values = {
 
         return _.isArray(subTypes) ? subTypes : [];
     },
+    hasSubTypes: function () {
+      return this.get('abstract') ||  this.get('eAllSubTypes');
```

Comment:
Also this does not seem correct. A class being abstract does not mean it has sub types. And the second test `this.get('eAllSubTypes')` is always true, it should be `this.get('eAllSubTypes').length`